### PR TITLE
Books

### PR DESCRIPTION
**Pull request by allistergg**

### CODE REVIEW

**Review by sbhuju61 — CHANGES_REQUESTED** [review]
I get error below on console
```
> Store does not have a valid reducer. Make sure the argument passed to combineReducers is an object whose values are reducers. 
>     in Books (created by ConnectFunction)
>     in ConnectFunction (created by Router.Consumer)
>     in Router.Consumer (created by Route)
>     in Route (at AppBody.js:15)
>     in AppBody (at App.js:18)
>     in App (at src/​index.js:13)
> console.<computed> @ index.js:1
> r @ react_devtools_backend.js:6
```

File: src/store/reducers/index.js [hunk 1]
```diff
@@ -30,4 +30,4 @@ export default combineReducers({
     [Target.PUBLISHER]  : publishersReducer,
     [Target.BRANCH]     : branchesReducer,
     [Target.BORROWER]   : borrowersReducer,
```
Comment (sbhuju61) [inline src/store/reducers/index.js:32]: other reducers need to be commented out. It is throwing error on console. 

export default combineReducers({
  [Target.BOOK]: booksReducer
});

File: src/index.js [hunk 2]
```diff
@@ -7,17 +7,11 @@ import 'primereact/resources/primereact.min.css';
...
 
-import store from './store';
-
 ReactDOM.render(
-    <Provider store={store}>
-        <App />
-    </Provider>
-    , document.getElementById('root'));
+ <App />
+   , document.getElementById('root'));
```
Comment (Jindanupajit-SS) [inline src/index.js]: You need a `<Provider store={store}>... </Provider>` or your `<App />` will not work properly.

File: src/components/Books.js [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                </DataTable>
+
+                <Dialog visible={this.state.displayDialog} width="300px" header="Book Details" modal={true} footer={dialogFooter} onHide={() => this.setState({ displayDialog: false })}>
+                    {
+                        this.state.book &&
+
+                        <div className="p-grid p-fluid">
+                            <div className="p-col-4" style={{ padding: '.75em' }}><label htmlFor="title">Title</label></div>
+                            <div className="p-col-8" style={{ padding: '.5em' }}>
+                                <InputText id="title" required={true} onChange={(e) => { this.updateProperty('title', e.target.value) }} value={this.state.book.title} />
```
Comment (sbhuju61) [inline src/components/Books.js]: disable save button when length is < 2. Currently it is sending request and getting error code 400 (BAD REQUEST) back.

File: src/store/reducers/index.js (hunk 1 above)
Comment (allistergg) [reply src/store/reducers/index.js:32]: Yeah, it's a warning, but should be gone once other reducers are filled in

File: src/index.js (hunk 2 above)
Comment (allistergg) [reply src/index.js]: The Provider is in App.js and seems to be working fine.

Comment (allistergg) [reply src/index.js]: Moving provider back to index.js...

**Review by bmasles — APPROVED** [review]

**Review by Jindanupajit-SS — APPROVED** [review]

**Review by sbhuju61 — APPROVED** [review]

**Pull request merged**